### PR TITLE
[vectorized](function) support order by field function

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.02 seconds
 load time: 438 seconds
 storage size: 17150093406 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221123142001_clickbench_pr_51669.html

File: docs/zh-CN/docs/sql-manual/sql-functions/string-functions/field.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+under the License.
+-->
+
+## field
+### description
+#### Syntax
+
+`field(Expr e, param1, param2, param3,.....)`
+
+在order by子句中，可以使用自定义排序，可以将expr中的数据按照指定的param1，2，3顺序排列。
```
Comment (HappenLee) [inline docs/zh-CN/docs/sql-manual/sql-functions/string-functions/field.md:33]: image

Comment (HappenLee) [reply docs/zh-CN/docs/sql-manual/sql-functions/string-functions/field.md:33]: Check the logic order by field (name 'banana', 'apple', 'banana') the logic of mysql. return error ?

File: be/src/vec/functions/least_greast.cpp [hunk 2]
```diff
@@ -110,17 +111,96 @@ struct CompareMultiImpl {
...
+        auto& res_data = static_cast<ColumnInt32*>(result_column)->get_data();
+
+        const auto& column_size = arguments.size();
+        ColumnPtr argument_columns[column_size];
+        for (int i = 0; i < column_size; ++i) {
+            argument_columns[i] = block.get_by_position(arguments[i]).column;
+        }
+        argument_columns[0] = argument_columns[0]->convert_to_full_column_if_const();
+        WhichDataType which(data_type);
+        if (which.is_string_or_fixed_string()) {
```
Comment (HappenLee) [inline be/src/vec/functions/least_greast.cpp:134]: ADD a TODO: hashmap to speed up the function

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**